common/remote/rmtssh.cpp

```diff
@@ -416,27 +444,36 @@ class CFRunSSH: public CInterface, implements IFRunSSH
                 expandCmd(cmdline,i,treefrom);
                 cmdline.append('"');
             }
-            if (dryrun) 
+            if (dryrun)
                 printf("%s\n",cmdline.str());
-            else {
+            else
+            {
                 Owned<IPipeProcess> pipe = createPipeProcess();
                 if (pipe->run((verbose&&!usepssh)?"FRUNSSH":NULL,cmdline.str(),workdir,
-                    useplink, // for some reason plink needs input handle
-                    true,true)) {
+                              useplink, // for some reason plink needs input handle
+                              true,true))
+                {
                     byte buf[4096];
-                    for (;;) {
+                    for (;;)
+                    {
                         size32_t read = pipe->read(sizeof(buf),buf);
                         if (!read)
                             break;
                         outbuf.append(read,(const char *)buf);
                     }
                     retcode = pipe->wait();
                     bool firsterr=true;
-                    for (;;) {
+                    for (;;)
+                    {
                         size32_t read = pipe->readError(sizeof(buf),buf);
                         if (!read)
                             break;
-                        if (firsterr) {
+
+                        if (!retcode)
+                            retcode = -3;
```

Comment:
I don't think we can assume that just because there's output on stderr the command has failed.
To be honest, I'm not sure what the correct etiquette is, but I think plenty of progrmas/scripts will produce stderr but not fail.
If a script thought it was a failure, it should capture the error and deal with, i.e. either ignore it, or fail and return an error code.

So I don't think this is the right fix.
We should only error if the command returns an error code.